apps/dashboard/src/lib/components/Form/TextField.svelte

```diff
@@ -57,56 +64,123 @@
   });
 </script>
 
-<label class="block relative {className}">
-  {#if label}
-    <p for="text-field" class="text-sm dark:text-white text-left m-0 {labelClassName}">
-      {label}
-
-      {#if isRequired}
-        <span class="text-red-700">*</span>
-      {/if}
-    </p>
-  {/if}
-  <input
-    use:typeAction
-    class="form-input border-l-0 border-r-0 border-t-0 border-b-2 border-gray-200 dark:border-neutral-600 focus:border-l-0 focus:border-r-0 rounded-t-md focus:border-t-0 focus:border-b-2 {inputClassName} {focusClass} {isDisabled &&
-      'hover:cursor-not-allowed opacity-50'} dark:text-black p-3 mt-1 block w-full {bgColor} {errorMessage
-      ? 'border-red-600'
-      : ''}"
-    on:keydown={onKeyDown}
-    on:change={onInputChange}
-    {placeholder}
-    bind:value
-    bind:this={ref}
-    required={isRequired}
-    disabled={isDisabled}
-    autocomplete={autoComplete ? 'on' : 'off'}
-    on:blur={(e) => {
-      if (focusClass.includes('border-primary-600')) {
-        focusClass = focusClass.replace('border-primary-600', '');
-      }
-      onChange(e);
-    }}
-    on:focus={() => (focusClass += ' border-primary-600')}
-    {name}
-    {min}
-    {max}
-  />
-  {#if isPassword}
-    <span class="password-eye">
-      <IconButton value="write-code" onClick={handlePasswordEye}>
-        <PasswordEye isClosed={type === 'password' ? true : false} />
+<div class="relative">
+  {#if showSearchBar && !toggleSearch}
+    <span>
+      <IconButton value="write-code" onClick={() => (toggleSearch = true)}>
```

Comment:
https://www.loom.com/share/5b657a4ff58340e9af86eb3d14f6de52?sid=a123e8d6-a1ee-48cf-a45c-ce967f90818a


i really only added it to match the design i saw, all we need to do to use the search option with the modal is this:
  showSearchBar={true} (this is false by default)
  bind:value={searchValue}

Reply:
I understand, you should probably have just passed a `slot` into that part instead of adding more props and logic into the component. I will refactor it myself